deployer/src/deployer/upload.py

```diff
@@ -369,18 +376,23 @@ def upload(
         content_roots,
         existing_bucket_objects=None,
         on_task_complete=None,
+        skip_redirects=False,
+        dry_run=False,
     ):
         with concurrent.futures.ThreadPoolExecutor(
             max_workers=MAX_WORKERS_PARALLEL_UPLOADS
         ) as executor, StopWatch() as timer:
             # Upload the redirects first, then the built files. This
             # ensures that a built file overrides its stale redirect.
-            for task_iter in (
-                lambda: self.iter_redirect_tasks(content_roots),
-                lambda: self.iter_file_tasks(build_directory),
-            ):
+            task_iters = []
+            if not skip_redirects:
+                task_iters.append(
+                    self.iter_redirect_tasks(content_roots, dry_run=dry_run)
+                )
+            task_iters.append(self.iter_file_tasks(build_directory, dry_run=dry_run))
```

Comment:
When writing this code originally, I had also called the generator functions immediately when creating the list of `task_iters`, because they'll return iterators (actually special generator iterators) so there's no need to wrap them in another layer of function call as lambdas (which as you see I did in the end). However, I found that doing that dramatically delayed when the progress bar appeared and displayed its results, which still remains a mystery to me. When we have time, we should discuss that because it still bothers me that I don't understand why it happens. Anyway, what you have here works fine, but does introduce that mysterious delay in the appearance of the progress bar.

Reply:
You're nerd-sniping me. I wanna understand that. 
But if it's a rabbit hole, I'll ignore it for now.